pkg/app/api/grpcapi/web_api.go

```diff
@@ -1500,3 +1503,92 @@ func (a *WebAPI) getInsightDataForChangeFailureRate(
 		Value:     changeFailureRate,
 	}, nil
 }
+
+// getInsightDataForMTTR accumulate insight data in target range for mean time to restore
+func (a *WebAPI) getInsightDataForMTTR(
+	ctx context.Context,
+	projectID string,
+	applicationID string,
+	targetRangeFrom time.Time,
+	targetRangeTo time.Time) (*model.InsightDataPoint, error) {
+
+	var applicationIDs []string
+	if applicationID == "" {
+		apps, err := a.applicationStore.ListApplications(ctx, datastore.ListOptions{})
```

Comment:
Why do we need to list the apps when the appID was not specified?
Doesn't an empty appID mean all apps of the project?

Reply:
Oh, I should have been corrected together when nakabonne reviewed. Thanks :D

Reply:
Simply my mistake. 
Since the original function `calculateAverageMTTR` was to calculate the MTTR of an application, I got all the appId's and calculated the MTTR by getting the deployment from the DB for each application in a loop to use calculateAverageMTTR around.
I changed calculateAverageMTTR to calculate MTTR from a mix of applications when I received nakabonne review, so I should have noticed that I don't need to get all the appId's here.